rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2110,6 +2110,148 @@ private static Stream<BridgeConstants> provideBridgeConstants() {
+        assertEquals(2, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        // Assert the two added pegouts has the same pegoutCreationRskTxHash
+        assertEquals(1,
+            provider
+                .getPegoutsWaitingForConfirmations()
+                .getEntries()
+                .stream()
+                .map(PegoutsWaitingForConfirmations.Entry::getPegoutCreationRskTxHash)
+                .collect(Collectors.toSet()).size()
+        );
+
+        // Advance blockchain to the height where both pegouts have enough confirmations to be moved to waitingForSignature
+        rskCurrentBlock = blockGenerator.createBlock(184, 1);
+        bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(rskCurrentBlock)
+            .withActivations(activations)
+            .build();
+
+        tx = Transaction
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
+        bridgeSupport.updateCollections(tx);
+
+        // Since only one pegout per updateCollection call is move to waitingForisgnatures
+        // -assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
+        assertEquals(1, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        assertEquals(1, provider.getPegoutsWaitingForSignatures().size());
+
+        // Advance blockchain one block to make the second pegout to be moved to waitingForConfirmation
+        rskCurrentBlock = blockGenerator.createBlock(185, 1);
+        BridgeSupport bridgeSupportForFailingTx = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(rskCurrentBlock)
+            .withActivations(activations)
+            .build();
+
+        Transaction throwsExceptionTx = Transaction
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
+
+        assertThrows(IllegalStateException.class, () -> {
+            bridgeSupportForFailingTx.updateCollections(throwsExceptionTx);
+        });
+
+
+        assertEquals(1, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        assertEquals(1, provider.getPegoutsWaitingForSignatures().size());
```

Comment:
Could we take this a bit further? Remove the migration from waiting for signatures (assume it got signed) and on the next call to updateCollections assert that the pending pegout is confirmed

Reply:
Done.